mqtt-client-Go/pubsub/test/mqttTest.go

```diff
@@ -46,18 +48,19 @@ func main() {
 				token := mqttClient.Publish(topic, 2, false, []byte(body))
 				if token.Error() != nil {
 					fmt.Printf("publish qos:2 topic:%s err:%v \n", topic, token.Error())
-				if !token.WaitTimeout(3 * time.Second) {
-					fmt.Printf("pub timeout:%d topic:%s \n", 3, topic)
+					if !token.WaitTimeout(3 * time.Second) {
+						fmt.Printf("pub timeout:%d topic:%s \n", 3, topic)
+					}
+					atomic.AddUint64(&ops, 1)
 				}
-				atomic.AddUint64(&ops, 1)
 			}
 		}()
-	}
-	for i := 0; i < 100000; i++ {
-		time.Sleep(time.Second * 1)
-		fmt.Println("pub and sub ------>>>", ops)
-	}
+		for i := 0; i < 100000; i++ {
```

Comment:
I don't understand the meaning of this code
Maybe you can consider trying the channel

Reply:
In the previous version, there was some syntax error in this file, I just made a simple modification to it